backend/common_utils/src/dapr.rs

```diff
@@ -0,0 +1,334 @@
+use anyhow::{Context as _, Result};
+use dapr::Client;
+use serde::{Deserialize, Serialize};
+use std::collections::HashMap;
+use tracing::info;
+use uuid::Uuid;
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct Event {
+    pub event_uuid: String,
+    pub timestamp: String,
+    pub flow_type: String,
+    pub connector: String,
+    pub stage: String,
+    pub latency: Option<u64>,
+    pub status_code: Option<u16>,
+    pub error_code: Option<String>,
+    pub error_reason: Option<String>,
+    pub request_data: Option<serde_json::Value>,
```

Comment:
Any untyped value should be secret and not as serde_json::Value this can be secretvalue